test/cases/azure.sh

```diff
@@ -0,0 +1,240 @@
+#!/bin/bash
+set -euo pipefail
+
+source /etc/os-release
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+#TODO: Remove this once there is rhel9 support for Azure image type
+if [[ $DISTRO_CODE == rhel_90 ]]; then
+    greenprint "Skipped"
+    exit 0
+fi
+
+# Provision the software under tet.
+/usr/libexec/osbuild-composer-test/provision.sh
+
+# Apply lorax patch to work around pytoml issues in RHEL 8.x.
+# See BZ 1843704 or https://github.com/weldr/lorax/pull/1030 for more details.
+if [[ $ID == rhel ]]; then
+    sudo sed -r -i 's#toml.load\(args\[3\]\)#toml.load(open(args[3]))#' \
+        /usr/lib/python3.6/site-packages/composer/cli/compose.py
+    sudo rm -f /usr/lib/python3.6/site-packages/composer/cli/compose.pyc
+fi
+
+# We need azure-cli to talk to Azure.
+if ! hash az; then
+    # this installation method is taken from the official docs:
+    # https://docs.microsoft.com/cs-cz/cli/azure/install-azure-cli-linux?pivots=dnf
+    sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+    echo -e "[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc" | sudo tee /etc/yum.repos.d/azure-cli.repo
+
+  greenprint "Installing azure-cli"
+  sudo dnf install -y azure-cli
+  az version
+fi
+
+# We need terraform to provision the vm in azure and then destroy it
+if [ "$ID" == "rhel" ] 
+then
+    release="RHEL"
+elif [ "$ID" == "fedora" ]
+then
+    release="fedora"
+else
+    echo "Test is not running on neither Fedora nor RHEL, terminating!"
+    exit 1
+fi
+sudo dnf install -y dnf-plugins-core
+sudo dnf config-manager --add-repo https://rpm.releases.hashicorp.com/$release/hashicorp.repo
+sudo dnf install -y terraform
+
+TEST_UUID=$(uuidgen)
+IMAGE_KEY=osbuild-composer-azure-test-${TEST_UUID}
+
+# Jenkins sets WORKSPACE to the job workspace, but if this script runs
+# outside of Jenkins, we can set up a temporary directory instead.
+if [[ ${WORKSPACE:-empty} == empty ]]; then
+    WORKSPACE=$(mktemp -d)
+fi
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+AZURE_CONFIG=${TEMPDIR}/azure.toml
+BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
+COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
+COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
+
+# Check for the smoke test file on the Azure instance that we start.
+smoke_test_check () {
+    SMOKE_TEST=$(sudo ssh -i key.rsa redhat@"${1}" -o StrictHostKeyChecking=no 'cat /etc/smoke-test.txt')
```

Comment:
Is `redhat` always the default username for ssh ? IIRC that was coming from cloud-init and in some older versions it was different between RHEL/Fedora (possibly CentOS). In the old lorax-composer test suite we're using `azure-user` instead but IIRC we weren't using cloud-init back then.

Reply:
Also in main.tf , line 163 you have `azureuser`. What's the relation with that ?

Reply:
`redhat` username comes from the cloud-init. It was `azureuser` by default but after I included the cloud-init file it changed to `redhat`... About the `azureuser` specified in `main.tf` right now I'm not even sure if that user exists there as I was unable to log in using that user... I can do some more investigation or just change the `azureuser` in `main.tf` to `redhat` and see if that works so that it's consistent.

Reply:
@ondrejbudai can you comment on that? I think we're fine as-is without further changes.

Reply:
Hmm, we are initializating the user twice - once in the terraform file and once by injecting a cloud-init. I think we should choose one. Also, the test seems to be failing on Fedora.

Reply:
Uhm, terraform requires to specify either `admin_ssh_key` or `admin_password` and I guess we want to use cloud-init for the smoke test. I could try creating another cloud-init file but I see very little reason to do so... Also the failure on Fedora is resolved, it was due to missing cloud-init package in the blueprint.